kratos/solving_strategies/schemes/residual_based_bossak_velocity_scheme.h

```diff
@@ -0,0 +1,714 @@
+                            const double OldAcceleration)
+    {
+        rCurrentDisplacement = OldDisplacement + mBossak.C6 * OldVelocity +
+                               mBossak.C4 * OldAcceleration + mBossak.C5 * CurrentAcceleration;
+    }
+
+    static void CalculateBossakConstants(BossakConstants& rBossakConstants,
+                                         const double Alpha,
+                                         const double DeltaTime)
+    {
+        TimeDiscretization::Bossak bossak(Alpha, 0.25, 0.5);
+        rBossakConstants.Alpha = bossak.GetAlphaM();
+        rBossakConstants.Gamma = bossak.GetGamma();
+        rBossakConstants.Beta = bossak.GetBeta();
+
+        rBossakConstants.C0 =
+            (1.0 - rBossakConstants.Alpha) / (rBossakConstants.Gamma * DeltaTime);
+        rBossakConstants.C1 =
+            DeltaTime / (rBossakConstants.Beta * rBossakConstants.Gamma);
+        rBossakConstants.C2 = 1.0 / (rBossakConstants.Gamma * DeltaTime);
+        rBossakConstants.C3 = (1.0 - rBossakConstants.Gamma) / rBossakConstants.Gamma;
+        rBossakConstants.C4 =
+            std::pow(DeltaTime, 2) * (-2.0 * rBossakConstants.Beta + 1.0) / 2.0;
+        rBossakConstants.C5 = std::pow(DeltaTime, 2) * rBossakConstants.Beta;
+        rBossakConstants.C6 = DeltaTime;
+    }
+
+    static void SynchronizeVariables(const std::vector<const VariableData*>& rVariables,
+                                     Communicator& rComm)
+    {
+        KRATOS_TRY;
+        for (auto p_variable_data : rVariables)
+        {
+            if (KratosComponents<Variable<array_1d<double, 3>>>::Has(
+                    p_variable_data->Name()))
+            {
+                const auto& r_variable =
+                    KratosComponents<Variable<array_1d<double, 3>>>::Get(
+                        p_variable_data->Name());
+                rComm.SynchronizeVariable(r_variable);
+            }
+            else if (KratosComponents<Variable<double>>::Has(p_variable_data->Name()))
+            {
+                const auto& r_variable =
+                    KratosComponents<Variable<double>>::Get(p_variable_data->Name());
+                rComm.SynchronizeVariable(r_variable);
+            }
+            else
+            {
+                KRATOS_ERROR << "Variable \"" << p_variable_data->Name() << "\" not found!\n";
+            }
+        }
+        KRATOS_CATCH("");
+    }
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+
+private:
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    typename TSparseSpace::DofUpdaterPointerType mpDofUpdater =
+        TSparseSpace::CreateDofUpdater();
+
+    BossakConstants mBossak;
+
+    std::vector<std::vector<IndirectScalar<double>>> mIndirectCurrentVelocityVector;
```

Comment:
what is an "IndirectScalar" ? I am lost here

Reply:
Sorry, myself @msandre changed the IndirectScalar today. This is used to get the references of derivatives from elements so that they could be updated in the scheme. At the moment, there is no way to pass references of first derivatives and second derivatives from the element. That's why we used this indirect scalar.

Reply:
could u expand a bit? with a small example maybe...